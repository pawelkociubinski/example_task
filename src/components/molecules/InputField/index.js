/* @flow */
// Dependencies
import React, { Component } from 'react';
import type { Node } from 'react';
// Custom
import Label from '../../atoms/Label';
import TextField from '../../atoms/TextField';

import { Container, Elevator } from './styles';

type Props = {
  label: string,
  onChange?: (event: Event) => void,
  readOnly?: boolean,
  value: string,
  icon?: Node,
};

type State = {
  active: boolean
};

class InputField extends Component<Props, State> {
  constructor() {
    super();
    this.state = {
      active: false,
    };
  }

  _handleBlur = (): void => {
    this.setState({ active: false });
  };

  _handleFocus = (): void => {
    this.setState({ active: true });
  };

  render() {
    const {
      value, onChange, readOnly, label, icon,
    } = this.props;
    const { active } = this.state;

    return (
      <Container>
        <Elevator lifted={active || value.length !== 0}>
          <Label theme={active ? 'active' : 'default'}>{label}</Label>
        </Elevator>
        <TextField
          active={active}
          onBlur={this._handleBlur}
          onChange={onChange}
          onFocus={this._handleFocus}
          readOnly={readOnly}
          value={value}
        />
        {icon}
      </Container>
    );
  }
}

export default InputField;
