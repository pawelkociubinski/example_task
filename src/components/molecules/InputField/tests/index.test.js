import React from 'react';
// import renderer from 'react-test-renderer';
import InputField from '../index';
import { shallow } from 'enzyme';


describe('<InputField />', () => {
  it('renders correctly', () => {
    const component = shallow(<InputField label="label" value="value" />);
    component.setState({ active: true });

    expect(component).toMatchSnapshot();
  });

  it('_handleFocus()', () => {
    const component = shallow(<InputField label="label" value="value" />);
    component.instance()._handleFocus();

    const state = component.state().active;

    expect(state).toEqual(true);
  });

  it('_handleBlur()', () => {
    const component = shallow(<InputField label="label" value="value" />);
    component.setState({ active: true });
    component.instance()._handleBlur();

    const state = component.state().active;

    expect(state).toEqual(false);
  });
});
