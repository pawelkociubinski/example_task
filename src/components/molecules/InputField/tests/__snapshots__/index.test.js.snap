// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputField /> renders correctly 1`] = `
<styled.div>
  <styled.div
    lifted={true}
  >
    <Label
      theme="active"
    >
      label
    </Label>
  </styled.div>
  <TextField
    active={true}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    readOnly={false}
    theme="neutral"
    value="value"
  />
</styled.div>
`;
